kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/checksum/MetadataChecksumGenerator.java

```diff
@@ -17,10 +19,13 @@ public interface MetadataChecksumGenerator {
     String CHECKSUM_CONTEXT_KEY = "kroxylicious.io/referent-checksum-generator";
     String NO_CHECKSUM_SPECIFIED = "";
 
-    static String checksumFor(HasMetadata... metadataSources) {
+    static String checksumFor(Stream<? extends HasMetadata> metadataSources) {
+        List<? extends HasMetadata> sources = metadataSources.toList();
```

Comment:
If you're just gonna call `toList()` on the stream maybe you should just make the parameter a `List<...>`. Passing around `Stream` always feels a little hairy, given the wide variety of behaviour they can support.

Reply:
yeah probably a bit dodgy since they're single use and the client has to know if you're going to consume it, will go for List